preflight_parser/preflight_parser/__init__.py

```diff
@@ -322,9 +343,11 @@ def detect_language(self) -> None:
             language = LanguageType.latex
         if re.search(r"\\text(bf|it|sl)|\\section|\\chapter", self._data, re.MULTILINE):
             language = LanguageType.latex
-        if re.search(r"^[^%\n]*\\bye[^a-zA-Z0-9_\n]", self._data, re.MULTILINE):
+        if re.search(r"^[^%\n]*\\bye[^a-zA-Z0-9_]", self._data, re.MULTILINE):
```

Comment:
Changed to the negative look-ahead. Indeed, a \bye without a final newline could happen.

Concenring documentstyle: This will be used only for new submissions, and I expect it to NOT use documentstyle. If it is, the user must manually select it.

Reply:
Pushed a commit that uses the negative lookahead and a unittest for it